### ISSUE TITLE
Espanso reports an error when a user/config.yml is empty

### ISSUE THREAD

**Issue by fstamour** [federico-terzi/espanso#600]
# Description

It's a small issue, but when I created a placeholder file in the `user` folder, espanso told me it was "Unable to reload the configuration due to an error. See logs fo" (yes the message was cropped in the notification) and got me searching around for an actual syntax error in the 3 YAML files I have. Until I looked at the command `espanso log` (which I didn't know about).

And `espanso log` gave me:
```
Error parsing YAML file '[...]\espanso\user\special-characters.yml', invalid syntax: EOF while parsing a value
```

# Reproduction step

Create an empty configuration file in the user folder.

## What I got

An error from espanso.

## What I would expect

No error. At worst, a warning.

## Notes

The configuration file just had comments in it when I created it, I was going to add rules later. I was able to reproduce the error with a completely empty file and with a file with just a "matches:". I had to add an actual rule for the error to go away.

Related to #589 "Think about ways to make invalid YAML syntax errors more clear and actionable".

Maybe add a context menu entry to open a window showing the latest error? My first "reflex" to find the log was to right-click on the system-tray icon, then to look in the same folder as the configurations and finally, I had to google how to see the logs. The notification didn't help _much_ because It's shown for like a second and it's only after a few times that I noticed the word "log" at the end of the notification message.

# P.S.

I just discovered espanso this week and it's awesome ❤️ . I've been looking for something like that for so long (free and open, simple (if you know YAML it is simple), cross-platform, etc).

**Comment by federico-terzi** [comment]
Hey @fstamour,

Thank you for the feedback! This is definitely an area that should be improved, and I'll make sure to include your suggestions as part of the #594 refactor (in which the config loading logic will be rewritten from scratch).

Cheers :)

**Comment by fstamour** [comment]
Nice https://github.com/federico-terzi/espanso/blob/2283cedbd352872c83ac8fdd89afe960dcfad315/espanso-config/src/util.rs#L20-L37

**Comment by federico-terzi** [comment]
@fstamour Indeed, the best thing about rewriting espanso from scratch is that I can fix a lot of long-standing problems!
